x/nft/keeper/grpc_query_test.go

```diff
@@ -0,0 +1,583 @@
+package keeper_test
+
+import (
+	gocontext "context"
+	"fmt"
+	"testing"
+
+	"github.com/stretchr/testify/require"
+	"github.com/stretchr/testify/suite"
+
+	"github.com/cosmos/cosmos-sdk/x/nft"
+)
+
+func TestGRPCQuery(t *testing.T) {
+	suite.Run(t, new(TestSuite))
+}
+
+func (suite *TestSuite) TestGRPCQueryBalance() {
+	var (
+		req     *nft.QueryBalanceRequest
+		balance uint64
+	)
+	testCases := []struct {
+		msg      string
+		malleate func(require *require.Assertions)
+		expError string
+		postTest func(require *require.Assertions, res *nft.QueryBalanceResponse)
+	}{
+		{
+			"fail empty ClassId",
+			func(require *require.Assertions) {
+				req = &nft.QueryBalanceRequest{}
+			},
+			"class id can not be empty",
+			func(require *require.Assertions, res *nft.QueryBalanceResponse) {},
+		},
+		{
+			"fail invalid Owner addr",
+			func(require *require.Assertions) {
+				req = &nft.QueryBalanceRequest{
+					ClassId: testClassID,
+					Owner:   "owner",
+				}
+			},
+			"decoding bech32 failed",
+			func(require *require.Assertions, res *nft.QueryBalanceResponse) {},
+		},
+		{
+			"Success",
+			func(require *require.Assertions) {
+				suite.TestMint()
+				req = &nft.QueryBalanceRequest{
+					ClassId: testClassID,
+					Owner:   suite.addrs[0].String(),
+				}
+				balance = 1
```

Comment:
Can w please add `balance` to test case structure? The `postTest` can either take balance as an argument or a pointer to the whole test case